api/processor_api.py

```diff
@@ -366,14 +424,88 @@ async def delete_area(area_id):
                 raise HTTPException(status_code=404, detail="Area does not exist")
 
             config_dict.pop(f'Area_{index}')
+            config_dict = reestructure_areas((config_dict))
+            self.config.update_config(config_dict)
+            self.config.reload()
+
+            # TODO: Restart only when necessary, and only the threads that are necessary (for instance to load a new video)
+            success = restart_processor()
+            return handle_config_response(config_dict, success)
+
+        @app.get("/cameras")
+        async def list_cameras(options: Optional[str] = ""):
+            return {
+                "cameras": get_cameras(options)
+            }
+
+        @app.get("/cameras/{camera_id}")
+        async def get_camera(camera_id):
+            cameras = get_cameras(['withImage'])
+            filtered_cameras = list(filter(lambda camera: camera['id'] == camera_id, cameras))
+            if not filtered_cameras:
+                raise HTTPException(status_code=404, detail=f'The camera: {camera_id} does not exist')
+            return filtered_cameras[0]
+
+        @app.post("/cameras")
+        async def create_camera(new_camera: SourceConfigDTO):
+            config_dict = extract_config()
+            cameras_name = [x for x in config_dict.keys() if x.startswith("Source")]
+            cameras = [map_camera(x, config_dict, []) for x in cameras_name]
+            if new_camera.id in [camera['id'] for camera in cameras]:
+                raise HTTPException(status_code=400, detail="Camera already exists")
+
+            config_dict[f'Source_{len(cameras)}'] = map_to_camera_file_format(new_camera)
+            self.config.update_config(config_dict)
+            self.config.reload()
+
+            # TODO: Restart only when necessary, and only the threads that are necessary (for instance to load a new video)
+            success = restart_processor()
+            return handle_config_response(config_dict, success)
+
+        @app.put("/cameras/{camera_id}")
+        async def edit_camera(camera_id, edited_camera: SourceConfigDTO):
+            edited_camera.id = camera_id
+            config_dict = extract_config()
+            cameras_names = [x for x in config_dict.keys() if x.startswith("Source")]
+            cameras = [map_camera(x, config_dict, []) for x in cameras_names]
+            cameras_ids = [camera['id'] for camera in cameras]
+            try:
+                index = cameras_ids.index(camera_id)
+            except ValueError:
+                raise HTTPException(status_code=404, detail="Camera does not exist")
```

Comment:
I think that we should send the same error message that we are sending in the `get`
```
detail=f'The camera: {camera_id} does not exist'
```